cumulusci/tasks/push/push_api.py

```diff
@@ -625,10 +620,8 @@ def _add_batch(self, batch, request_id):
                         "INVALID_OPERATION",
                         "UNKNOWN_EXCEPTION",
                     ]:
-                        org_id = self._get_org_id(
-                            batch_data["records"], result["referenceId"]
-                        )
-                        invalid_orgs.add(org_id)
+                        org_id = result["referenceId"]
+                        batch.remove(org_id)
```

Comment:
So I didn't write this code, but I attempted to revert the code however it is failing tests @davisagli do you mind discussing with me further?

Reply:
@jofsky I just added a commit to fix it. One of the tests relied on the orgs getting passed to the API in a specific order, which means it's not possible to store them in a set, since sets don't have a stable order.